central/notifiers/all/all_test.go

```diff
@@ -0,0 +1,56 @@
+package all
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path"
+	"path/filepath"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var blacklist = []string{"all", "utils", "validation", "metadatagetter"}
+
+func TestAllPackagesAreImported(t *testing.T) {
+	notifiersDirEntry, err := os.ReadDir("..")
+	require.NoError(t, err, "failed to read notifiers directory")
+
+	existingNotifiers := set.NewStringSet()
+	for _, entry := range notifiersDirEntry {
+		if !entry.IsDir() {
+			continue
+		}
+		baseName := filepath.Base(entry.Name())
+
+		if slices.Contains(blacklist, baseName) {
+			continue
+		}
+		existingNotifiers.Add(baseName)
+	}
+
+	var allImports []*ast.ImportSpec
+	f, err := parser.ParseFile(token.NewFileSet(), "all.go", nil, parser.ImportsOnly)
+	require.NoError(t, err, "failed to parse all.go")
+
+	allImports = append(allImports, f.Imports...)
+
+	importedNotifiers := set.NewStringSet()
+	for _, imp := range allImports {
+		pkgName := strings.TrimSuffix(strings.TrimPrefix(imp.Path.Value, `"`), `"`)
+		pkgBaseName := path.Base(pkgName)
+		importedNotifiers.Add(pkgBaseName)
+	}
+
+	existingButNotImported := existingNotifiers.Difference(importedNotifiers)
+	importedButNotExisting := importedNotifiers.Difference(existingNotifiers)
```

Comment:
Wouldn't this fail to compile anyways?

Reply:
It's possible by importing a package outside of the `notifiers` package.

Reply:
In this case the dev is required to add the package to the whitelist. (Needed to rename blacklist to white list, got the name wrong.)